eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/RouteBlinding.scala

```diff
@@ -114,58 +120,67 @@ object RouteBlindingEncryptedDataCodecs {
   private val nextBlinding: Codec[NextBlinding] = (("length" | constant(hex"21")) :: ("blinding" | publicKey)).as[NextBlinding]
   private val paymentRelay: Codec[PaymentRelay] = (
     ("length" | constant(hex"0a")) ::
-    ("fee_base_msat" | millisatoshi32) ::
+      ("fee_base_msat" | millisatoshi32) ::
       ("fee_proportional_millionths" | uint32) ::
       ("cltv_expiry_delta" | cltvExpiryDelta)).as[PaymentRelay]
   private val paymentConstraints: Codec[PaymentConstraints] = variableSizeBytesLong(varintoverflow,
     ("max_cltv_expiry" | cltvExpiry) ::
       ("htlc_minimum_msat" | millisatoshi) ::
       ("allowed_features" | featuresCodec)).as[PaymentConstraints]
 
-
-  private val messageRelayDataTlvCodec = discriminated[RouteBlindingEncryptedDataTlv].by(varint)
+  private val encryptedDataTlvCodec = discriminated[RouteBlindingEncryptedDataTlv].by(varint)
     .typecase(UInt64(1), padding)
+    .typecase(UInt64(2), outgoingChannelId)
     .typecase(UInt64(4), outgoingNodeId)
+    .typecase(UInt64(6), pathId)
     .typecase(UInt64(8), nextBlinding)
+    .typecase(UInt64(10), paymentRelay)
+    .typecase(UInt64(12), paymentConstraints)
```

Comment:
I really don't see this as a readability improvement, these two codecs only had two TLVs in common (padding being there only for testing purposes) and now you need to explicitly disallow a bunch of TLVs in the narrowed codecs.

Reply:
It is a readability improvement: when you read the spec you see a list of tlvs that can be included in onion payloads. Then you look at the `eclair` code and you see the same list and can easily check that everything that's in the spec is supported by `eclair`. In the previous version it was split into several codecs, there were more types in the type hierarchy and a `fallback(fallback(X, Y), Z)` which is very hacky...

> you need to explicitly disallow a bunch of TLVs in the narrowed codecs.

I only kept that code because it was in your version, but I'm really in favor of removing it.
It's completely unnecessary: what do we care if the sender wasted space by putting some tlvs that we'll simply ignore?
This is extra code that doesn't protect us more than simply ignoring these fields.
The only benefit I see is to provide senders with a nice error message to figure out where they messed up, but it doesn't make sense in route blinding since we will never return a sender-readable error.